src/Core/Domain/Product/Image/Command/ProductImageSetting.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Image\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\ValueObject\ImageId;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopId;
+
+class ProductImageSetting
+{
+    /**
+     * @var ImageId
+     */
+    private $productImageId;
+    /**
+     * @var array<ShopId>
+     */
+    private $shopIds;
+
+    public function __construct(int $productImageId, array $shopIds)
```

Comment:
also add docblock here because it would be nice to have $shopIds array structure documented with psalm notation here too